src/Microsoft.ML.Data/Transforms/ValueMappingTransformer.cs

```diff
@@ -153,6 +153,23 @@ private static ValueGetter<VBuffer<ReadOnlyMemory<char>>> GetKeyValueGetter<TKey
                 };
         }
 
+        internal static void AddColumnWrapper<T>(ArrayDataViewBuilder builder, string columnName, PrimitiveType primitiveType, T[] values)
+        {
+            if (typeof(T) == typeof(string))
+                builder.AddColumn(columnName, values.Select(x=>x.ToString()).ToArray());
+            else
+                builder.AddColumn(columnName, primitiveType, values);
+        }
+
+        internal static void AddColumnWrapper<T>(ArrayDataViewBuilder builder, string columnName, PrimitiveType primitiveType, T[][] values)
+        {
+            if (typeof(T) == typeof(string))
+
+                builder.AddColumn(columnName, primitiveType, values.Select(x => x.Select( y => y.ToString().AsMemory() ).ToArray()).ToArray());
```

Comment:
>x => x.Select( y => y.ToString().AsMemory() ).ToArray()) [](start = 75, length = 56)

Please try avoiding recursive structure in one line.

Reply:
Its now expanded into a few more lines...

---
In reply to: [247217068](https://github.com/dotnet/machinelearning/pull/2115#discussion_r247217068) [](ancestors = 247217068)